### PR TITLE
add more info to "protocol mux failed"

### PR DESCRIPTION
**Pull request by zhiqiangxu**
We also met [this issue](https://github.com/libp2p/go-libp2p/issues/339), it may help to provide more info besides "protocol mux failed"

### CODE REVIEW

**Comment by sukunrt** [comment]
You probably want to improve the error message in go-multistream. `s.Protocol()` will be empty string here since the protocol selection failed.

**Comment by zhiqiangxu** [comment]
> You probably want to improve the error message in go-multistream. `s.Protocol()` will be empty string here since the protocol selection failed.

It seems [`MultistreamMuxer.Negotiate`](https://github.com/multiformats/go-multistream/blob/master/multistream.go#L194)  don't have too much info since its argument is `ReadWriteCloser`. I removed `Protocol` since it's always empty.

File: p2p/host/basic/basic_host.go [hunk 1]
```diff
@@ -414,7 +414,7 @@ func (h *BasicHost) newStreamHandler(s network.Stream) {
 			}
 			logf("protocol EOF: %s (took %s)", s.Conn().RemotePeer(), took)
 		} else {
-			log.Debugf("protocol mux failed: %s (took %s)", err, took)
+			log.Debugf("protocol mux failed: %s (took %s, id:%s)", err, took, s.ID())
```
Comment (sukunrt) [inline p2p/host/basic/basic_host.go:417]: We can add details from the connection here. 
`s.Conn().RemotePeer()` and `s.Conn().RemoteMultiAddr()`

Comment (zhiqiangxu) [reply p2p/host/basic/basic_host.go]: Done.

**Review by sukunrt — APPROVED** [review]
Thanks @zhiqiangxu

**Comment by zhiqiangxu** [comment]
It seems the ci failure isn't related to this PR?

**Comment by MarcoPolo** [comment]
No worries, I'll rerun. We have some [flaky tests](https://marcopolo.github.io/FlakyTests/) I need to debug.

**Pull request merged**